axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/bankreconciliation/BankReconciliationService.java

```diff
@@ -340,15 +342,52 @@ public Move generateMove(
             description,
             bankReconciliationLine.getBankReconciliation().getBankDetails());
 
-    generateMoveLine(bankReconciliationLine, bankStatementRule, move, true);
-    moveLineTaxService.autoTaxLineGenerate(move);
+    MoveLine moveLine = generateMoveLine(bankReconciliationLine, bankStatementRule, move, true);
+
+    generateTaxMoveLine(moveLine, bankStatementRule);
+
+    moveLine = generateMoveLine(bankReconciliationLine, bankStatementRule, move, false);
 
-    MoveLine moveLine = generateMoveLine(bankReconciliationLine, bankStatementRule, move, false);
     bankReconciliationLineService.reconcileBRLAndMoveLine(bankReconciliationLine, moveLine);
 
     return moveRepository.save(move);
   }
 
+  protected void generateTaxMoveLine(MoveLine moveLine, BankStatementRule bankStatementRule)
+      throws AxelorException {
+    int vatSystemSelect = AccountRepository.VAT_SYSTEM_DEFAULT;
+    Move move = moveLine.getMove();
+    int journalTechnicalType = move.getJournal().getJournalType().getTechnicalTypeSelect();
+    Company company = moveLine.getMove().getCompany();
+    if (journalTechnicalType == JournalTypeRepository.TECHNICAL_TYPE_SELECT_EXPENSE) {
+      AccountingSituation accountingSituation =
+          accountingSituationRepository.findByCompanyAndPartner(company, moveLine.getPartner());
+      if (accountingSituation != null && accountingSituation.getVatSystemSelect() != null) {
+        vatSystemSelect = accountingSituation.getVatSystemSelect();
+      }
+    } else if (journalTechnicalType == JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE) {
+      AccountingSituation accountingSituation =
+          accountingSituationRepository.findByCompanyAndPartner(company, company.getPartner());
+      if (accountingSituation != null && accountingSituation.getVatSystemSelect() != null) {
+        vatSystemSelect = accountingSituation.getVatSystemSelect();
+      }
```

Comment:
Duplicated code, just because the partner is different, so just set in a variable the partner depending on the journalTechnicalType value and then do this code